### PR TITLE
Contacts, Limits, Group Chat Filtering, and clean up Column types

### LINKED ISSUE #39: Add Filtering Functions to TopFriends Query
Add the logic to filter on the TopFriends query just like the WordOrEmoji query.

### PR DESCRIPTION
**Pull request by alexdanilowicz**
- Add filtering logic for contacts, group chat filtering
- Add OutputColumns to make it clear what columns can be referenced. 

Resolves #39  and https://github.com/Left-on-Read/app/issues/6

Note: I'm quite pleased the queries are extremely fast :) 

Screen Shot 2021-01-24 at 10 32 31 PM

### CODE REVIEW

**Comment by alexdanilowicz** [comment]
@Teddarific care to review? I realized keeping tracking of all the application state in that `<Root>` component (eventually to be renamed `<Dashboard>`) may become unscalable, but for now, I thought lifting the state up was a fine solution considering we don't have to prop-drill to the center of the earth. (I was inspired by this _excellent_ blog post: https://kentcdodds.com/blog/application-state-management-with-react)

File: electron/src/components/charts/WordOrEmojiCountChart.tsx [hunk 1]
```diff
@@ -10,30 +10,27 @@ interface WordOrEmojiCountProps {
...
+        const data = await chatBro.queryEmojiOrWordCounts(db, filters);
         setWords(data.map((obj) => obj.word));
         setCount(data.map((obj) => obj.count));
       } catch (err) {
         log.error(`ERROR fetching for component for ${titleText}`, err);
       }
     }
     fetchWordData();
-  }, []);
+  }, [db, titleText, filters]);
```
Comment (ayc19) [inline electron/src/components/charts/WordOrEmojiCountChart.tsx:33]: what does this do

**Review by ayc19 — APPROVED** [review]
nice super clean

File: electron/src/components/charts/WordOrEmojiCountChart.tsx (hunk 1 above)
Comment (alexdanilowicz) [reply electron/src/components/charts/WordOrEmojiCountChart.tsx:33]: https://dev.to/rozenmd/understanding-useeffect-the-dependency-array-obg

^ that will explain it better than i ever could

**Pull request merged**